libsplinter/src/circuit/template/mod.rs

```diff
@@ -78,34 +78,33 @@ impl CircuitTemplateManager {
 
     pub fn list_available_templates(&self) -> Result<Vec<String>, CircuitTemplateError> {
         let path = Path::new(&self.path);
-
-        path.read_dir()
-            .map_err(|err| {
-                CircuitTemplateError::new_with_source(
-                    &format!("Failed to read files in {}", self.path),
-                    Box::new(err),
-                )
-            })?
-            .map(|entry| {
-                let file = entry.map_err(|err| {
-                    CircuitTemplateError::new_with_source(
-                        &format!("Failed to read file in {}", self.path),
-                        Box::new(err),
-                    )
-                })?;
-                Ok(file
-                    .file_name()
-                    .into_string()
-                    .map_err(|_| {
-                        CircuitTemplateError::new(&format!(
-                            "Failed to read file name {}",
-                            self.path
-                        ))
-                    })?
-                    .trim_end_matches(".yaml")
-                    .to_string())
-            })
-            .collect::<Result<_, CircuitTemplateError>>()
+        let mut available_templates: Vec<String> = vec![];
+        match path.read_dir() {
+            Ok(entries) => {
+                for entry in entries {
+                    match entry {
+                        Ok(file) => match file.file_name().into_string() {
+                            Ok(name) => {
+                                available_templates
+                                    .push(name.trim_end_matches(".yaml").to_string());
+                            }
+                            Err(_) => {
+                                error!("Unable to read file name {}", self.path);
+                            }
+                        },
+                        Err(err) => {
+                            error!("Unable to read file {}", err);
```

Comment:
These error logs should provide a bit more context, like "Unable to read circuit template {}". Otherwise they're pretty much unidentifiable errors.

Reply:
Good point, fixed